### ISSUE TITLE
No Websocket url within 2 seconds

### ISSUE THREAD

**Issue by daBee** [rubycdp/ferrum#56]
Fresh install on **MacOS El Capitan**.  I tried the Google connection, and other sites, and this error pops up:

`/Users/rich/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/ferrum-0.7/lib/ferrum/browser/process.rb:137:in `parse_ws_url': Browser process did not produce websocket url within 2 seconds (RuntimeError)`

I'm not sure what to provide at this point.  `/usr/local/bin/chromedriver` is installed.

**Comment by route** [comment]
Hi @daBee You don't need `chromedriver` btw, try to increase `Ferrum::Browser.new(process_timeout: 5)` to 5 seconds maybe your laptop isn't that fast..

**Comment by route** [comment]
Let me know if that didn't work for you, but it should be that easy

**Comment by daBee** [comment]
Hi there.  Sorry, but github was barfing when I tried to answer.  Here's what I tried to post:

OK, leaving that `chromedriver` out I get that error:

`Cliver::Dependency::NotFound (Could not find an executable for the browser. Try to make it available on the PATH or set environment varible for example BROWSER_PATH="/usr/bin/chrome")`

Leaving this in, at 10 seconds I still don't get a websocket:

`ENV['BROWSER_PATH'] = '/usr/local/bin/chromedriver'`

**Comment by route** [comment]
It should be a path to Chrome not ChromeDriver, if you are on mac it's something like:

`BROWSER_PATH="/Applications/Google Chrome.app/Contents/MacOS/Google Chrome"`

**Comment by daBee** [comment]
Ah I escaped that path before and it didn't work.  **NOT** escaping it worked.  Excellent.  Thank you.

**Comment by oshanz** [comment]
how I solved it
- add `gem 'webdrivers', '~> 4.0'` to Gemfile 
- on script `require 'webdrivers/chromedriver'`

**Comment by route** [comment]
@oshanz this is wrong you don't need `webdrivers` at all!

**Comment by oshanz** [comment]
hmm yes I'm getting error again randomly

**Comment by route** [comment]
Set `process_timeout` to 5 from https://github.com/rubycdp/ferrum#customization

**Comment by senaxor** [comment]
Hi there, just having this error on my rails app running on server using Ferrum browser 
*Browser did not produce websocket url within 10 seconds, try to increase `:process_timeout`*
i tried to increase process_timeout to 50 but nothing changes.

the code on server: 
browser = Ferrum::Browser.new(browser_path: "/snap/chromium/current/usr/lib/chromium-browser/chrome",browser_options: { 'no-sandbox': nil })

**Comment by senaxor** [comment]
thanks for the response,
I just fixed it with adding xvfb:true option and also this command was needed 
// RUN apt-get install -qy xvfb
 then I changed google-chrome to google-chrome-stable.
 


> Maybe you don't have enough memory on the server. Check your server, the issue is definitely there.

**Comment by Tashows** [comment]
In my case, my app was running in a container that was built with paketo buildpacks. I had to add `libnspr4`, `libnss3`, `libatk-bridge2.0-0`, `libxkbcommon0`, `libasound2`, `libgbm1` in the Aptfile of my project, along with `https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb`. Didn't have to mess with any other options on the `Ferrum::Browser` call, it just worked. I figured out that's what was needed by exec-ing into the container and trying to run the chrome executable.
